lockapi.pm

```diff
@@ -141,8 +142,17 @@ sub barrier_wait (@args) {
             return 1;
         }
 
-        bmwqemu::diag("barrier '$name' not released, sleeping " . POLL_INTERVAL . ' seconds');    # uncoverable statement
-        sleep POLL_INTERVAL;    # uncoverable statement
+        my $poll_interval = POLL_INTERVAL;
+        if ($timeout != -1) {
+            my $elapsed_time = time - $start;
+            if ($elapsed_time >= $timeout) {
+                bmwqemu::mydie "barrier '$name' timeout after $elapsed_time seconds";
+            } else {
+                $poll_interval = $timeout - $elapsed_time if ($timeout - $elapsed_time < $poll_interval);
```

Comment:
Just a suggestion:
```suggestion
                $poll_interval = min($timeout - $elapsed_time, $poll_interval);
```
It would require `use List::Util qw(min);`.